src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -135,7 +135,23 @@ private bool ArgumentsEquals(IReadOnlyList<Type> genericArguments)
             return true;
         }
 
-        public override int GetHashCode() => HashCode.Combine(_implementation, _type, ((System.Collections.IStructuralEquatable)Arguments).GetHashCode(EqualityComparer<CSharpType>.Default));
+        private int? _hashCode;
```

Comment:
Please add a unit test in Autorest.Csharp.Tests to ensure regressions like this are caught.

Reply:
tests are added